plugin/listeners.py

```diff
@@ -26,6 +48,7 @@ def _is_saving(self) -> bool:
     def _is_saving(self, value: bool) -> None:
         set_copilot_view_setting(self.view, "_is_saving", value)
 
+    @_must_be_active_view()
```

Comment:
Would it work to use `self.view.is_primary()`?

Checking if the view is active sounds rather brittle. What if view is not active? You'd potentially miss some update?

Reply:
If we use `self.view.is_primary()`, then when the user is typing in a non-primary view, how to make the popup appear in the non-primary view?

Reply:
If it's just about whether to show the popup then I guess it's fine.
I'm just thinking that the document can get modified when in the background but maybe that's not a problem here.

Reply:
> I'm just thinking that the document can get modified when in the background but maybe that's not a problem here.

Yes. If it's modified in background, I think it makes no sense to fire an Copilot request for it.

Reply:
@jfcherng I've been using this for awhile now with no foreseeable issues. It may be improved upon however at the moment I think this is working in a better respect that in the current release